channeldb/channel.go

```diff
@@ -1572,7 +1579,40 @@ func (c *OpenChannel) isBorked(chanBucket kvdb.RBucket) (bool, error) {
 // republish this tx at startup to ensure propagation, and we should still
 // handle the case where a different tx actually hits the chain.
 func (c *OpenChannel) MarkCommitmentBroadcasted(closeTx *wire.MsgTx,
-	locallyInitiated bool) error {
+	lFCInsight *LocalForceCloseInsights) error {
+
+	var locallyInitiated bool
```

Comment:
```suggestion
	locallyInitiated := lFCInsight != nil
```

Reply:
Is this a better way? Or just different, everything that has to do with when lFCInsight being nil, is  done in the following "If" conditional.

Reply:
If you can express something without mutation, you should. Mutation is the leading cause of unintended behavior because it makes logical structure dependent on the relative positions of statements and the compiler will not be able to catch bugs that arise from this.

Reply:
I am not sure if this is critically concerning. We would be duplicating logic if we follow this route. We literally check if lfcinsight is nil right after this.

Reply:
I gave you a [pair of buttons](https://github.com/lightningnetwork/lnd/pull/8072#discussion_r1417969099) that if you accept those suggestions together would fix the concern I have. So if you want you can find other people who want to do the review and approve it as is, but candidly I think it's more work to resist this than it is to just do it. If you think I'm *wrong*, I'm all ears, but the argument you gave me is that this isn't bad enough to warrant changing, which is not really an argument I'm going to accept.

There's more to software engineering in an open source environment than simply the code being functionally correct. Code is first and foremost about describing how the system works *to other developers* and only secondarily about describing it *to the machine*. There are many ways to write code that works in a way that is not resilient to the human element that sits beneath the software itself.

LND is responsible for handling money and the consequences of it being wrong are irrevocable. The code you wrote works, but it is not resilient to the "revolving door" nature that is Open Source development. Making this change reduces the chances that a developer who alters this code in the future does so in a way that creates bugs. Mutation, in general, is brittle. Sometimes (though still rarely) it is impractical to avoid. Here, it is straightforward to avoid, so we should.

Reply:
Thanks for the review. I also raised a concern that following this route would mean duplicating logic that literally pops up again in the next line. If this is a trade off that you suggest we are willing to make then I can go ahead with your suggestion.

Reply:
> Thanks for the review. I also raised a concern that following this route would mean duplicating logic that literally pops up again in the next line. If this is a trade off that you suggest we are willing to make then I can go ahead with your suggestion.

Then rewrite the next line to be `if locallyInitiated { ... }`. I agree this would be an improvement.